From f67ea3c62f33f0e675b3e091044c4983ae01b085 Mon Sep 17 00:00:00 2001
From: Thomas A Caswell <tcaswell@gmail.com>
Date: Tue, 6 Oct 2015 00:16:50 -0400
Subject: [PATCH] FIX: mro issue with pyside/pyqt4

Do not let a `super` be called or the the c++ base-classes will be
initialized twice which causes havoc.

 - pyside failed on figure creation
 - pyqt4 may segfault on exit

closes #5196
---
 lib/matplotlib/backends/backend_qt4agg.py | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/lib/matplotlib/backends/backend_qt4agg.py b/lib/matplotlib/backends/backend_qt4agg.py
index 392d5d5..d36991c 100644
--- a/lib/matplotlib/backends/backend_qt4agg.py
+++ b/lib/matplotlib/backends/backend_qt4agg.py
@@ -15,7 +15,7 @@
 from matplotlib.figure import Figure
 
 
-from .backend_qt5agg import FigureCanvasQTAggBase
+from .backend_qt5agg import FigureCanvasQTAggBase as _FigureCanvasQTAggBase
 
 from .backend_agg import FigureCanvasAgg
 from .backend_qt4 import QtCore
@@ -54,6 +54,11 @@ def new_figure_manager_given_figure(num, figure):
     return FigureManagerQT(canvas, num)
 
 
+class FigureCanvasQTAggBase(_FigureCanvasQTAggBase):
+    def __init__(self, figure):
+        self._agg_draw_pending = False
+
+
 class FigureCanvasQTAgg(FigureCanvasQTAggBase,
                         FigureCanvasQT, FigureCanvasAgg):
     """
